### PR TITLE
RELEASE-PROCEDURE: update to new schedule

### PR DESCRIPTION
**Pull request by bagder**
Ref: https://curl.se/mail/lib-2023-03/0062.html

Attempted illustration of the new setup:

curl release cycle(1)

### CODE REVIEW

File: docs/RELEASE-PROCEDURE.md [hunk 1]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
...
+arise, we can insert releases outside of the schedule or we can move the
+release date.
 
-Each 8 week release cycle is split in two 4-week periods.
+Each 8 week (56 days) release cycle is split in three distinct periods.
 
-- During the first 4 weeks after a release, we allow new features and changes
-  to curl and libcurl. If we accept any such changes, we bump the minor number
-  used for the next release.
+- During the first calendar 10 days after a release, we are in "cool down". We
```
Comment (MarcelRaad) [inline docs/RELEASE-PROCEDURE.md:75]: Should this be `10 calendar days` instead of `calendar 10 days`?

Comment (andy5995) [inline docs/RELEASE-PROCEDURE.md:75]:
```suggestion
- During the first 10 calendar days after a release, we are in "cool down". We
```

File: docs/RELEASE-PROCEDURE.md [hunk 2]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
...
 
-We do releases every 8 weeks on Wednesdays. If critical problems arise, we can
-insert releases outside of the schedule or we can move the release date - but
-this is rare.
+We normally do releases every 8 weeks on Wednesdays. If critical problems
+arise, we can insert releases outside of the schedule or we can move the
+release date.
 
-Each 8 week release cycle is split in two 4-week periods.
+Each 8 week (56 days) release cycle is split in three distinct periods.
```
Comment (andy5995) [inline docs/RELEASE-PROCEDURE.md:73]:
```suggestion
Each 8 week (56 days) release cycle is split into three distinct periods.
```

or "divided by" ?

File: docs/RELEASE-PROCEDURE.md [hunk 3]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
...
 
-- During the first 4 weeks after a release, we allow new features and changes
-  to curl and libcurl. If we accept any such changes, we bump the minor number
-  used for the next release.
+- During the first calendar 10 days after a release, we are in "cool down". We
+  do not merge features but only bug-fixes. If any important enough regression
```
Comment (andy5995) [inline docs/RELEASE-PROCEDURE.md:76]:
```suggestion
  do not merge features but only bug-fixes. If a critical regression
```

File: docs/RELEASE-PROCEDURE.md [hunk 4]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
...
-  release.
+- During the following 3 weeks (21 days) there is a feature window: we allow
+  new features and changes to curl and libcurl. If we accept any such changes,
+  we bump the minor number used for the next release.
 
-- After a regular procedure-following release (made on Wednesdays), the
-  feature window remains closed until the following Monday in case of special
-  actions or patch releases etc.
+- During the next 25 days we are in feature freeze. We do not merge any
+  features or changes, we then only focus on fixing bugs and polishing things
```
Comment (andy5995) [inline docs/RELEASE-PROCEDURE.md:84]:
```suggestion
  features or changes, and we only focus on fixing bugs and polishing things
```

File: docs/RELEASE-PROCEDURE.md [hunk 5]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
...
+  to make a solid coming release.
 
 If a future release date happens to end up on a "bad date", like in the middle
 of common public holidays or when the lead release manager is away traveling,
```
Comment (andy5995) [inline docs/RELEASE-PROCEDURE.md:88]:
```suggestion
of common public holidays or when the lead release manager is unavailable,
```

File: docs/EARLY-RELEASE.md [hunk 6]
```diff
@@ -8,10 +8,11 @@ away".
...
 ## Bugfix
 
-During the release cycle, and especially in the beginning of a new cycle,
-there are times when a bug is reported and after it has been subsequently
-fixed correctly, the discussion might be brought up: is this bug and
-associated fix important enough for an early patch release.
+During the release cycle, and especially in the beginning of a new cycle (the
+co called "cool down" period), there are times when a bug is reported and
+after it has been subsequently fixed correctly, the discussion might be
+brought up: is this bug and associated fix important enough for an early patch
```
Comment (andy5995) [inline docs/EARLY-RELEASE.md:14]:
```suggestion
asked: is this bug and associated fix important enough for an early patch
```

File: docs/EARLY-RELEASE.md [hunk 7]
```diff
@@ -8,10 +8,11 @@ away".
...
+release.
 
 The question can only be properly asked once a fix has been created and landed
```
[the remark this answers is not in the document]
Comment (andy5995) [reply docs/EARLY-RELEASE.md:17]: or "after". "Once" isn't bad - the meaning is clear, but I felt the wording could be slightly better.

File: docs/EARLY-RELEASE.md [hunk 8]
```diff
@@ -8,10 +8,11 @@ away".
...
 
-During the release cycle, and especially in the beginning of a new cycle,
-there are times when a bug is reported and after it has been subsequently
-fixed correctly, the discussion might be brought up: is this bug and
-associated fix important enough for an early patch release.
+During the release cycle, and especially in the beginning of a new cycle (the
+co called "cool down" period), there are times when a bug is reported and
+after it has been subsequently fixed correctly, the discussion might be
+brought up: is this bug and associated fix important enough for an early patch
+release.
```
[the remark this answers is not in the document]
Comment (andy5995) [reply docs/EARLY-RELEASE.md:15]: >the discussion might be
brought up: is this bug and associated

It sounded more like a question to me, so I suggested changes in 3 different parts (including adding a '?'). If it's a discussion, you may want to change the phrasing: "the discussion might be brought up to decide whether the bug and associated fix are important..."

File: docs/EARLY-RELEASE.md [hunk 9]
```diff
@@ -28,9 +29,10 @@ big and we never release just a patch. There is only "release".
  - Is there a security advisory rated high or critical?
```
Comment (dfandrich) [inline docs/EARLY-RELEASE.md:29]: A high security advisory doesn't necessarily mean an immediate patch release unless the flaw is publicly known. But, I suppose it is "a question to ask".

File: docs/EARLY-RELEASE.md [hunk 10]
```diff
@@ -28,9 +29,10 @@ big and we never release just a patch. There is only "release".
...
  - Did the bug cause an API/ABI breakage?
+ - Will the problem annoy an non-insignificant share of the user population?
 
-If the answer is yes to one or more of the above, an early release might
-indeed be warranted.
+If the answer is yes to one or more of the above, an early release might be
+warranted.
 
 More questions to ask ourselves when doing the assessment if the answers to
 the three ones above are all 'no'.
```
Comment (dfandrich) [inline docs/EARLY-RELEASE.md:38]: (this comment is for line 52 but GitHub doesn't let me leave it there). Change "How long is it left" to "How long is it" (or "How long is left").

(line 60) If we're lowering the bar for patch releases. "Unless done for security or similarly important reasons" seems to strongly worded. And "two weeks" should be now "10 days".

File: docs/RELEASE-PROCEDURE.md [hunk 11]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
 Release Cycle
 -------------
 
-We do releases every 8 weeks on Wednesdays. If critical problems arise, we can
-insert releases outside of the schedule or we can move the release date - but
-this is rare.
+We normally do releases every 8 weeks on Wednesdays. If critical problems
```
Comment (dfandrich) [inline docs/RELEASE-PROCEDURE.md:69]: "critical" is now too strong.

File: docs/RELEASE-PROCEDURE.md [hunk 12]
```diff
@@ -66,23 +66,23 @@ curl release scheduling
...
+arise, we can insert releases outside of the schedule or we can move the
```
Comment (dfandrich) [inline docs/RELEASE-PROCEDURE.md:70]: Drop "of"

**Comment by bagder** [comment]
I've tried to incorporate all the suggestions. Take another look and see if there are details left to polish. I included a fancy image in the top description.

**Review by dfandrich — APPROVED** [review]
Looks good! You can drop the 8.0.0 release mentioned under Coming date. Also, "time" is singular in "Round and round until the end of time…".

**Comment by pheiduck** [comment]
@bagder The Everything curl book still contains the old graphic.
https://github.com/bagder/everything-curl/blob/master/project/release-cycle.png

**Pull request closed without merging**